test/external-modules/apm-integration/src/javaRestTest/java/org/elasticsearch/test/apmintegration/ApmIT.java

```diff
@@ -43,19 +52,100 @@ public void testApmIntegration() throws Exception {
         // TestMeterUsages for code that is registering & using metrics
         assertBusy(() -> {
             var usageRecords = mockApmServer.getMessages();
-            assertThat(
-                usageRecords,
-                Matchers.hasItems(
-                    containsString("\"testDoubleCounter\":{\"value\":1.0}"),
-                    containsString("\"testLongCounter\":{\"value\":1.0}"),
-                    containsString("\"testDoubleHistogram\":{\"values\":[0.8535535000000001,1.7071049999999999],\"counts\":[1,1]"),
-                    // long is represented as doubles in apm server
-                    containsString("\"testLongHistogram\":{\"values\":[0.8535535000000001,1.7071049999999999],\"counts\":[1,1]"),
-                    containsString("\"testDoubleGauge\":{\"value\":1.0}"),
-                    containsString("\"testLongGauge\":{\"value\":1}")
-                )
-            );
-
+            var maybeUsages = usageRecords.stream().map(APMMessage::new).reduce(APMMessage::merge);
+            assertTrue(maybeUsages.isPresent());
+            var usages = maybeUsages.get();
+            assertThat(usages.getDouble("testDoubleCounter"), closeTo(1.0, 0.001));
+            assertThat(usages.getDouble("testLongCounter"), closeTo(1.0, 0.001));
+            assertThat(usages.getDouble("testDoubleGauge"), closeTo(1.0, 0.001));
+            assertThat(usages.getLong("testLongGauge"), equalTo(1L));
+            assertThat(usages.getHistogram("testDoubleHistogram").stream().mapToInt(Bucket::count).sum(), equalTo(2));
```

Comment:
Nit: I would use a closeTo here as well, to avoid false positives due to math (2.0000000001)

Reply:
I think it should be safe. counts are represented as integers 
```
"samples":{"testDoubleHistogram":{"values":[0.8535535000000001,1.7071049999999999],"counts":[1,1],"type":"histogram"}
```

Reply:
Ah sorry, I was looking at the wrong fields there. Of course with integers is fine.